src/common/task.cc

```diff
@@ -348,7 +359,93 @@ const std::unordered_map<std::string, double> TaskSpec_get_required_resources(
   return map_from_flatbuf(*message->required_resources());
 }
 
-bool TaskSpec_is_dependent_on(TaskSpec *spec, ObjectID object_id) {
+TaskSpec *TaskSpec_copy(TaskSpec *spec, int64_t task_spec_size) {
+  TaskSpec *copy = (TaskSpec *) malloc(task_spec_size);
+  memcpy(copy, spec, task_spec_size);
+  return copy;
+}
+
+void TaskSpec_free(TaskSpec *spec) {
+  free(spec);
+}
+
+TaskExecutionSpec::TaskExecutionSpec(
+    const std::vector<ObjectID> &execution_dependencies,
+    TaskSpec *spec,
+    int64_t task_spec_size) {
+  execution_dependencies_ = execution_dependencies;
+  task_spec_size_ = task_spec_size;
+  TaskSpec *spec_copy = new TaskSpec[task_spec_size_];
+  memcpy(spec_copy, spec, task_spec_size);
```

Comment:
By the way, I would suggest switching to `std::copy` or `std::copy_n`, since that's the way to copy memory in C++. This code will break if at any time the data you are copying becomes nontrivial to copy (such as if the data type gains any members with constructors or destructors).